airflow/providers/openlineage/conf.py

```diff
@@ -145,3 +145,8 @@ def execution_timeout() -> int:
     """[openlineage] execution_timeout."""
     option = conf.get(_CONFIG_SECTION, "execution_timeout", fallback="")
     return _safe_int_convert(str(option).strip(), default=10)
+
+
+@cache
+def include_full_task_info() -> bool:
+    return _is_true(conf.get(_CONFIG_SECTION, "include_full_task_info", fallback="False"))
```

Comment:
we have `getboolean` ... I think it can be used here?

Reply:
Yes, probably it could, but i think we wanted to prevent breaking Airflow / raising errors by all means, so we put this explanation in the module docstring:
```
To prevent errors caused by invalid user-provided configuration values, we use ``conf.get()``
to fetch values as strings and perform safe conversions using custom functions.

Any invalid configuration values should be treated as incorrect and replaced with default values.
For example, if the default for boolean ``custom_ol_var`` is False, any non-true value provided:
``"asdf"``, ``12345``, ``{"key": 1}`` or empty string, will result in False being used.

By using default values for invalid configuration values, we ensure that the configurations are handled
safely, preventing potential runtime errors due to conversion issues.
```

Does it make sense or do you see any other viable way to do it?

Reply:
Well. It makes a bit inconsistent behaviour with other boolean flags, which might be surprising for the users. And if there is no good reason to diverge from the common approach, I think consistency trumps convenience (if you think the approach you proposed is a bit more convenient in some cases):

Is there a good reason why the defaulkt `getboolean` behaviour is a problem and can't be used ?

I mean this:

```python
    def getboolean(self, section: str, key: str, **kwargs) -> bool:  # type: ignore[override]
        val = str(self.get(section, key, _extra_stacklevel=1, **kwargs)).lower().strip()
        if "#" in val:
            val = val.split("#")[0].strip()
        if val in ("t", "true", "1"):
            return True
        elif val in ("f", "false", "0"):
            return False
        else:
            raise AirflowConfigException(
                f'Failed to convert value to bool. Please check "{key}" key in "{section}" section. '
                f'Current value: "{val}".'
            )
```

Reply:
BTW. I think the behaviour where invalid values are silently converted to default is a bad idea. Imagine this typo:

```
include_full_task_info: trve 
```

Which might be difficult to spot depending on the font you are using in your IDE

Reply:
Yes, what you're saying makes sense and it probably would be even easier for the user to debug (the whole Airflow stops working unless you provide a valid value). I don't know, i thought our approach makes sense but maybe we should change it back. Any thoughts @mobuchowski @JDarDagran ? I think I proposed it as we tried as hard as possible not to break anything in Airflow deployments, even if somebody provided an OL config with a typo.

Reply:
Ok, I've updated the PR to use `getboolean` consistently with rest of Airflow. I've converted rest in followup PR: https://github.com/apache/airflow/pull/40780

Reply:
Upd: @kacpermuda prepared better PR at https://github.com/apache/airflow/pull/40790